prov/hook/profile/include/prof_config.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2018-2023 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL); Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef _PROF_CONFIG_H_
+#define _PROF_CONFIG_H_
+
+#define HOOK_PROF_IGNORE_SIZE   0xffffffffffffffff
+#define STR_LEN 64
+
+/* top level class */
+enum {
+	HOOK_PROF_API,
+	HOOK_PROF_CQDATA,
+	HOOK_PROF_CLASS_MAX
+};
+
+struct profile_context;
+struct profile_cntr_class;
+struct profile_cntr_group;
+
+typedef void (*prof_update_counter_fn)(struct profile_context *ctx,
+               struct profile_cntr_class *cls,
+               struct profile_cntr_group *grp,
+               int cntr_idx, uint64_t value);
+
+typedef void (*prof_report_fn)(struct profile_context *ctx,
+              struct profile_cntr_class *cls);
+
+struct profile_event {
+	uint32_t value;
```

Comment:
This is the event value, for message-size events, it is the upper bound for each bucket.  Is 32bit (4GB) larger  enough?  I will change it to <event, name> to avoid confusion.